nemo/collections/asr/data/audio_to_target.py

```diff
@@ -0,0 +1,157 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Dict, Optional
+
+import torch
+
+from nemo.collections.asr.parts import collections
+from nemo.core.classes import Dataset
+from nemo.core.neural_types import AudioSignal, LengthsType, NeuralType, RegressionValuesType
+from nemo.utils import logging
+
+__all__ = ['AudioToTargetDataSet']
+
+
+class AudioToTargetDataSet(Dataset):
+    """Data Layer for general speech regression.
+    Module which reads speech recognition with a numeric target. It accepts comma-separated
+    JSON manifest files describing the correspondence between wav audio files
+    and their target values. JSON files should be of the following format::
+        {"audio_filepath": path_to_wav_0, "duration": time_in_sec_0, "target": \
```

Comment:
Where is `target` and `scalar` used in side `__getitem__`? This dataset seems to be a duplicate of the AudioToLabelDataSet

I think it would be best to reuse functionality in the preexisting dataset classes (by making them adaptive to the input data).

So if the AudioToLabelDataSet gets a `label` - assume its a classification problem. If its a `regression_target` - assume its a regression value.